### PR TITLE
Added Anti-armour Launcher

### PR DESCRIPTION
**Pull request by Thunder12345**
http://puu.sh/sxmHj/14ea75bddf.ogg
http://puu.sh/sxmz4/8e89001980.webm

:cl: Thunder12345
add: Added anti-armour launcher. A single-use rocket launcher capable of penetrating all but the heaviest of armour. Deals massively increased damage to cyborgs and mechs.
/:cl:

### CODE REVIEW

**Comment by GunHog** [comment]
Post a picture so I do not have to open the DMI file!

**Comment by Thunder12345** [comment]
Going one better, will post a webm of it in action as soon as it's ready

**Comment by Shadowlight213** [comment]
Add tanks. What am I supposed to shoot an anti tank gun at if there's no tanks?

**Comment by ma44** [comment]
This is essentially a single shot sniper rifle that can always delimb every shot, so you can price it somewhat higher than a sniper but make the ammo cheap for unknown reasons.

**Comment by Thunder12345** [comment]
Single-use is not the same as single shot though, it's one shot then throw away the useless empty tube

**Comment by KorPhaeron** [comment]
Seems a bit redundant as an anti-tank weapon when they have an ion rifle already

**Comment by Cyberboss** [comment]
Bigger is better

**Comment by Supermichael777** [comment]
ion rifle should probably be removed from them and they just get a free one of these. ion is after all an energy weapon and those are mostly NT.

also need to be able to make a horribly dangerous knockoff one with a fire extinguisher, a length of pipe, a metal rod, and some black powder.

**Comment by ToadStar** [comment]
Thing is, this synchronizes well with the ion rifle. The damage this does should do about 300+ damage to a Durand, and half of a charge worth of shots from the Ion Rifle would finish the Durand off. Plus it instantly kills Cyborgs, so if secborgs ever get turned back on, this is a hilarious overcompensation to deal with them that the ion rifle just doesnt give.

Oh, and my price suggestion is 5 TC, since it's effectively less powerful of a boom then a C4/X4, while being only a single use weapon makes it... Less viable if it were to cost any more. Or else it might end up being as unused as the other weapons were when they costed a truckload more.

**Comment by AnturK** [comment]
Needs bigger inhands.

**Comment by Shapsy** [comment]
Ion rifle is anti-energy though so it makes sense that of all the weapons the syndicate use that you can stick in a charger, that's the one

Also, ion rifles already stun borgs for a long time with each shot (and does so with a splash) and kills them pretty quickly, and crew combat mechs should be the last thing ops are worrying about when they usually take at least 30 minutes to build, and can usually be countered by the ion rifle anyway. The only thing that'd really be worth using an AT rifle on would be one of the operative mechs to begin with, really

**Comment by TechnoAlchemisto** [comment]
What you could do is do a minor retheme, have it do huge amounts of damage to everything and penetrate walls/structures ala tunguska meteor.

**Comment by Shadowlight213** [comment]
Nah, instead of making it simulate a tunguska, make it fire tunguskas. :^)

**Comment by KorPhaeron** [comment]
Yeah, the only faction in the game that has serious "tanks" is being given an anti tank launcher because ????

If this is ever used it will be for the one hit crit and aoe knockout, not because the ops are getting steamrolled by a ripley or egg borg

**Comment by ma44** [comment]
Even with it being one shot, it still causes a decent explosion, will delimb, pierces ALL armor and does 80 damage. This is pretty close to a "lol i shoot and u ded in 1 hit"

**Comment by Thunder12345** [comment]
Right now I'd rather just get it in game and worry about where to use it, so would it be better if I made it admin only initially, then we can discuss who, if anyone, to give it to later

**Comment by optimumtact** [comment]
Make it two handed

**Comment by Thunder12345** [comment]
Two handed things are their own separate child of /weapon, I've been demanding a refactor for them for ages now, but it's beyond me to actually do it

**Comment by Iamgoofball** [comment]
@Thunder12345 sniper rifles

**Comment by Thunder12345** [comment]
Sniper rifles aren't actually two handed they just fly out your hand if you shoot them with something in your other hand

File: code/modules/projectiles/projectile/special.dm [hunk 1]
```diff
@@ -43,6 +43,27 @@
...
+/obj/item/projectile/bullet/a84mm/on_hit(atom/target, blocked = 0)
+	..()
+	explosion(target, -1, 1, 3, 1, 0, flame_range = 4)
+
+	if(istype(target, /obj/mecha))
+		var/obj/mecha/M = target
+		M.take_damage(anti_armour_damage)
+	if(istype(target, /mob/living/silicon))
+		var/mob/living/silicon/S = target
+		S.take_overall_damage(3*anti_armour_damage/4, anti_armour_damage/4)
```
Comment (ChangelingRain) [inline code/modules/projectiles/projectile/special.dm]: Just do `anti_armour_damage * 0.75` for the first arg and  `anti_armour_damage * 0.25` for the second, it's easier to read/understand(and you shouldn't be dividing where you can avoid it)

Comment (Thunder12345) [inline code/modules/projectiles/projectile/special.dm]: This should've been blatantly obvious and I don't know what I was on to do that in the first place

**Comment by Supermichael777** [comment]
you can make it two handed by making it weight class 4 i believe.

**Comment by Thunder12345** [comment]
No that just makes it too big for backpacks

**Comment by optimumtact** [comment]
There is a weight flag of some kind you can toggle, I'm sure of it

**Comment by optimumtact** [comment]
weapon_weight = WEAPON_HEAVY

File: code/modules/projectiles/projectile/special.dm (hunk 1 above)
Comment (Shadowlight213) [inline code/modules/projectiles/projectile/special.dm]: Nonsense. Mso says he does his best coding work while high.

Comment (optimumtact) [inline code/modules/projectiles/projectile/special.dm]: best? he does all of his work that way

**Comment by Cebasiz** [comment]
it's kind of a shame the crew will never have mechs to actually kill and also kind of a shame that mechs aren't very good

**Comment by Supermichael777** [comment]
a battle mech koolaidmaning the armory used to be standered tryhard mode for ops. it was terible

File: code/modules/projectiles/projectile/special.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/projectiles/projectile/special.dm]: It's a matter of level of stoned

**Pull request merged**